platform-hub-api/app/controllers/kubernetes/groups_controller.rb

```diff
@@ -1,10 +1,95 @@
 class Kubernetes::GroupsController < ApiJsonController
 
+  before_action :find_group, only: [ :show, :update, :destroy ]
+
+  skip_authorization_check only: [ :index, :show ]
+  authorize_resource class: KubernetesGroup, :except => [ :index, :show ]
+
+  # GET /kubernetes/groups
+  def index
+    groups = KubernetesGroup.order(:name)
+    render json: groups
+  end
+
+  # GET /kubernetes/groups/:id
+  def show
+    render json: @group
+  end
+
+  # POST /kubernetes/groups
+  def create
+    group = KubernetesGroup.new(group_params)
+
+    if group.save
+      AuditService.log(
+        context: audit_context,
+        action: 'create',
+        auditable: group
+      )
+
+      render json: group, status: :created
+    else
+      render_model_errors group.errors
+    end
+  end
+
+  # PATCH/PUT /kubernetes/groups/:id
+  def update
+    if @group.update(group_params)
+      AuditService.log(
+        context: audit_context,
+        action: 'update',
+        auditable: @group
+      )
+
+      render json: @group
+    else
+      render_model_errors @group.errors
+    end
+  end
+
+  # DELETE /kubernetes/groups
+  def destroy
+    id = @group.id
+    name = @group.name
+
+    @group.destroy
+
+    AuditService.log(
+      context: audit_context,
+      action: 'destroy',
```

Comment:
As discussed maybe worth storing @groups as auditable for future filtering?

Reply:
👍